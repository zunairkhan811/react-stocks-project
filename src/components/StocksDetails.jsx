import { useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import details from '../redux/stocksdetails/DetailsApi';

const StocksDetails = () => {
  const { symbol } = useParams();
  const dispatch = useDispatch();
  const {
    data, isError, message, isLoading,
  } = useSelector((state) => state.stocksDetails);
  useEffect(() => {
    dispatch(details(symbol));
  }, [dispatch, symbol]);

  if (isLoading) {
    return (
      <div className="list-heading-container">
        <h1>{message}</h1>
      </div>
    );
  }
  if (isError) {
    return (
      <div className="list-heading-container">
        <h1>{message}</h1>
      </div>
    );
  }

  return (
    <>
      <div className="list-heading-container">
        <h1>
          {symbol}
          {' '}
          Stock Details
        </h1>
      </div>
      <div className="details-container">
        {data.map((item) => (
          <ul key={item.symbol}>
            <li>
              <h3>
                Symbol
              </h3>
              <h3>
                {item.symbol}
              </h3>
            </li>
            <li>
              <h3>
                Name
              </h3>
              <h3>
                {item.name}
              </h3>
            </li>
            <li>
              <h3>
                Exchange
              </h3>
              <h3>
                {item.exchange}
              </h3>
            </li>
            <li>
              <h3>
                Price
              </h3>
              <h3>
                {item.price}
              </h3>
            </li>
            <li>
              <h3>
                MarketCap
              </h3>
              <h3>
                {item.marketCap}
              </h3>
            </li>
            <li>
              <h3>
                Day-High
              </h3>
              <h3>
                {item.dayHigh}
              </h3>
            </li>
            <li>
              <h3>
                Day-Low
              </h3>
              <h3>
                {item.dayLow}
              </h3>
            </li>
            <li>
              <h3>
                Volume
              </h3>
              <h3>
                {item.volume}
              </h3>
            </li>
            <li>
              <h3>
                Year-High
              </h3>
              <h3>
                {item.yearHigh}
              </h3>
            </li>
            <li>
              <h3>
                Year-Low
              </h3>
              <h3>
                {item.yearLow}
              </h3>
            </li>
          </ul>

        ))}
      </div>

    </>

  );
};

export default StocksDetails;
