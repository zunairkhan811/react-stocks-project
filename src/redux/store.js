import { configureStore } from '@reduxjs/toolkit';
import rootReducer from './rootreducer';

const store = configureStore({
  reducer: rootReducer,
});

export default store;
