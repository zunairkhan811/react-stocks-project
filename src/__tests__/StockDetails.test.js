import renderer from "react-test-renderer";
import StocksDetails from "../components/StocksDetails";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../redux/store";
import '@testing-library/jest-dom';

it('StockDetails renders correctly',()=>{
    const tree = renderer.create(
        
        <BrowserRouter>
        <Provider store={store}>
        <StocksDetails />
        </Provider>
            
        </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})