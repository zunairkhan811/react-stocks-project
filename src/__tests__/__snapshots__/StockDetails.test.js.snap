// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockDetails renders correctly 1`] = `
Array [
  <div
    className="list-heading-container"
  >
    <h1>
       
      Stock Details
    </h1>
  </div>,
  <div
    className="details-container"
  />,
]
`;
